lib/web/app/handler.go

```diff
@@ -388,6 +388,11 @@ func (h *Handler) getAppSession(r *http.Request) (ws types.WebSession, err error
 		h.logger.WarnContext(r.Context(), "Failed to get session", "error", err)
 		return nil, trace.AccessDenied("invalid session")
 	}
+
+	if ws.Expiry().Before(h.c.Clock.Now()) {
+		h.logger.WarnContext(r.Context(), "Session expired")
+		return nil, trace.AccessDenied("session expired")
```

Comment:
I'm not sure what the implications are about revealing that a session is still accessible in the backend but expired, but we're not disclosing details about the specific error we hit when obtaining the session, so I reckon it's better to be consistent here.

```suggestion
		return nil, trace.AccessDenied("invalid session")
```

Reply:
Should the check be in `getAppSessionFromCert` and `getAppSessionFromCookie`? Is this the only place where we're fetching an `AppSession` where we should error out if it's expired?

Reply:
added a helper and updated other places calling accesspoint.GetAppSession. Updated the return error to `invalid session` and downgraded the log to debug.

10ab463